Amplo/AutoML/DataProcessor.py

```
@@ -577,24 +585,102 @@ def convert_float_int(self, data: pd.DataFrame = None) -> pd.DataFrame:
+        labels = pd.Series(labels)
+
+        # It's probably a classification task
+        if labels.dtype == object or labels.nunique() <= labels.size / 2:
+            # Create encoding
+            encoder = LabelEncoder()
+            if fit is True:
+                # Fit
+                encoder.fit(labels)
+                self._label_encodings = pd.Series(encoder.classes_).to_list()
+            elif not self._label_encodings:
+                raise NotFittedError('Encoder it not yet fitted')
+            else:
+                encoder.classes_ = np.ndarray(self._label_encodings)
+            # Encode
+            return encoder.transform(labels)
+
+        # It's probably a regression task, thus no encoding needed
+        if warn_unencodable:
+            warnings.warn(UserWarning('Labels are probably for regression. No encoding happened...'))
+        return labels.to_numpy()
+
+    def decode_labels(self, labels, *, except_not_fitted=True):
+        """Decode labels from numerical dtype to original value
+
+        Parameters
+        ----------
+        labels : np.ndarray or pd.Series
+            Labels to decode
+        except_not_fitted : bool
+            Whether to raise an exception when label encoder is not fitted
+
+        Returns
+        -------
+        np.ndarray
+            Decoded labels or original labels if label encoder is not fitted and `except_not_fitted` is True
+
+        Raises
+        ------
+        NotFittedError
+            When `except_not_fitted` is True and label encoder is not fitted
+        """
+        try:
+            if len(self._label_encodings) == 0:
+                raise NotFittedError('Encoder it not yet fitted. Try first calling `encode_target` '
+                                     'to set an encoding')
+            encoder = LabelEncoder()
+            encoder.classes_ = np.array(self._label_encodings)
+            return encoder.inverse_transform(labels)
+        except NotFittedError as err:
```

In what case is a `NotFittedError` thrown when the `self._label_encodings` is unknown?

---

I assume there is no other situation than that

---

Then this second clause is entirely redundant?

---

No, I don't think so.
The `except_not_fitted` parameter is here to purposely ignore the error and return the original labels. By default it's `False` so a user realizes that it does not make sense to "decode" the labels. However, I thought it is convenient for the function `_code_target_column()` such that it can handle both, regression and classification tasks.

---

If it makes more sense, I could change it a little bit that both, `encode_labels` and `decode_labels` raise an error if the labels resemble a regression and catch the error in `_code_target_column` instead.

---

Ah I see. You catch your own exception 🙈 . Then nvm, read the code wrong. All good :)